Workout Core/Model/Workout/Workout.swift

```diff
@@ -519,4 +520,169 @@ public class Workout {
+                    case self.paceRow:
+                        description += NSLocalizedString("WRKT_AVG_PACE", comment: "Average Pace")
+                        description += NSLocalizedString("WRKT_EVENT_SEPARATOR", comment: "Separator")
+                        description += (self.pace?.formatAsPace(withReferenceLength: self.paceUnit.unit(for: systemOfUnits)) ?? missingValueStr) + "\n"
+                    case self.speedRow:
+                        description += NSLocalizedString("WRKT_AVG_SPEED", comment: "Average Speed")
+                        description += NSLocalizedString("WRKT_EVENT_SEPARATOR", comment: "Separator")
+                        description += (self.speed?.formatAsSpeed(withUnit: self.speedUnit.unit(for: systemOfUnits)) ?? missingValueStr) + "\n"
+                    case self.energyRow:
+                        description += NSLocalizedString("WRKT_ENERGY", comment: "Energy")
+                        description += NSLocalizedString("WRKT_EVENT_SEPARATOR", comment: "Separator")
+                        if let total = self.totalEnergy {
+                            if let active = self.activeEnergy {
+                                description += String(format: NSLocalizedString("WRKT_SPLIT_CAL_%@_TOTAL_%@", comment: "Active/Total"),
+                                                           active.formatAsEnergy(withUnit: WorkoutUnit.calories.unit(for: systemOfUnits)),
+                                                           total.formatAsEnergy(withUnit: WorkoutUnit.calories.unit(for: systemOfUnits)))  + "\n"
+                            } else {
+                                description += total.formatAsEnergy(withUnit: WorkoutUnit.calories.unit(for: systemOfUnits)) + "\n"
+                            }
+                        } else {
+                            description += missingValueStr + "\n"
+                        }
+                    case self.elevationRow:
+                        description += NSLocalizedString("WRKT_ELEVATION", comment: "Elevation Change")
+                        description += NSLocalizedString("WRKT_EVENT_SEPARATOR", comment: "Separator")
+                        
+                        let (asc, desc) = self.elevationChange
+                        for (v, dir) in [(asc, "↗ "), (desc, "↘ ")] {
+                            guard let v = v else {
+                                continue
+                            }
+                                                        
+                            description += dir + v.formatAsElevationChange(withUnit: WorkoutUnit.elevation.unit(for: systemOfUnits)) + "\n"
+                        }
+                default:
+                    break
+                }
+            }
+            description.removeLast()
+            
+            eventStore.requestAccess(to: .event, completion: { (granted, error) in
```

Comment:
I think authorization to use the calendar should be handled by the main app, not `Workout Core`. Also the string containing the reason we use the calendar should be updated to reflect the new functionality, the current value is just there to satisfy a requirement from Apple but never actually used